pkg/controller/azvolume.go

```diff
@@ -250,68 +245,59 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 		if volumeDeleteRequested {
 			w.Logger().V(5).Info("Deleting Volume...")
 		}
+	}
 
-		waitCh := make(chan goSignal)
-		//nolint:contextcheck // call is asynchronous; context is not inherited by design
-		go func() {
-			_, goWorkflow := workflow.New(ctx)
-			var deleteErr error
-			defer func() { goWorkflow.Finish(deleteErr) }()
-			waitCh <- goSignal{}
-
-			goCtx := goWorkflow.SaveToContext(context.Background())
-
-			deleteCtx, deleteCancel := context.WithTimeout(goCtx, cloudTimeout)
-			defer deleteCancel()
+	waitCh := make(chan goSignal)
+	//nolint:contextcheck // call is asynchronous; context is not inherited by design
+	go func() {
```

Comment:
Do we need to do this in a goroutine?

Reply:
I don't see a reason for why we shouldn't. Otherwise, we will have a reconciler worker blocked waiting for the attachments to be deleted.

Reply:
I am not sure I understand what the waitChannel is used here for.

Reply:
This is to ensure that the caller exits after goroutine child workflow is created. Otherwise, we end up in a situation where parent workflow completes before child workflow is created.

Reply:
Oh, I see it now!